File: poultry-pal-service/src/main/java/co/za/hlaluko/dynamics/poultry/pal/model/dto/Expense.java
package co.za.hlaluko.dynamics.poultry.pal.model.dto;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Expense {
  private String id;
  private Date expenseDate;
  private String expenseType;
  private Double amount;
  private String additionalInfo;
  private String recordedBy;
  private Date createdDate;
}

File: poultry_pal_plus_app/lib/models/expense.dart
class Expense {
  final String id;
  final String expenseDate;
  final String expenseType;
  final double amount;
  final String additionalInfo;
  final String recordedBy;
  final String createdDate;

  Expense({
    required this.id,
    required this.expenseDate,
    required this.expenseType,
    required this.amount,
    required this.additionalInfo,
    required this.recordedBy,
    required this.createdDate,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      expenseDate: json['expenseDate'],
      expenseType: json['expenseType'],
      amount: json['amount'],
      additionalInfo: json['additionalInfo'] ?? '',
      recordedBy: json['recordedBy'],
      createdDate: json['createdDate'],
    );
  }
}
